pkg/plugin/processor/builtin/unwrap/unwrap_opencdc.go

```diff
@@ -0,0 +1,293 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=opencdc_paramgen.go openCDCConfig
+
+package unwrap
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/goccy/go-json"
+)
+
+type openCDCConfig struct {
+	// Field is a reference to the field which contains the OpenCDC record.
+	//
+	// For more information about record references, see: https://github.com/ConduitIO/conduit-processor-sdk/blob/cbdc5dcb5d3109f8f13b88624c9e360076b0bcdb/util.go#L66.
+	Field string `json:"field" validate:"regex=^.Payload" default:".Payload.After"`
+}
+
+type openCDCProcessor struct {
+	sdk.UnimplementedProcessor
+
+	logger      log.CtxLogger
+	fieldRefRes sdk.ReferenceResolver
+}
+
+func NewOpenCDCProcessor(logger log.CtxLogger) sdk.Processor {
+	return &openCDCProcessor{logger: logger}
+}
+
+func (u *openCDCProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "unwrap.opencdc",
+		Summary: "A processor that unwraps the OpenCDC record saved in one of record's fields.",
+		Description: `
+The unwrap.opencdc processors unwraps the OpenCDC record saved in one of the record's fields.
+
+This is useful in situations where a record goes through intermediate systems before 
+being written to a final destination. In these cases, the original OpenCDC record is
+part of the payload read from the intermediate system and needs to be unwrapped before
+being written to the destination.
+'`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: openCDCConfig{}.Parameters(),
+	}, nil
+}
+
+func (u *openCDCProcessor) Configure(ctx context.Context, m map[string]string) error {
+	cfg := openCDCConfig{}
+	err := sdk.ParseConfig(ctx, m, &cfg, cfg.Parameters())
+	if err != nil {
+		return cerrors.Errorf("failed parsing configuration: %w", err)
+	}
+
+	rr, err := sdk.NewReferenceResolver(cfg.Field)
+	if err != nil {
+		return cerrors.Errorf("invalid reference: %w", err)
+	}
+
+	u.fieldRefRes = rr
+	return nil
+}
+
+func (u *openCDCProcessor) Open(context.Context) error {
+	return nil
+}
+
+func (u *openCDCProcessor) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		proc := u.processRecord(rec)
```

Comment:
You know what I'll suggest here, right? 😄 
```suggestion
		proc, err := u.processRecord(rec)
```

Reply:
Ah, I missed changing this processor.